File: js/oop.js
class InputPrework{
    constructor(input){
        this.input = String(input);
        this.inputClass = "?";
        console.log("### item {item} 1 - INITIALIZED");
        this.setDataClass();
    }

    notInt(this){
        let notInt = False;
        for (let i = 0; i < len(this.input); i++) {
            let char = this.charAt(i);;
            if (char == ".") {
                notInt = True;
            }
        }
        return notInt;
    }

    notFloat(this){
        let notFloat = False;
        for (let i = 0; i < this.input.length; i++){
            let char = this.input.charAt(i);
            if ("-" == input.charAt(i) ||
                "0" == input.charAt(i) ||
                "1" == input.charAt(i) ||
                "2" == input.charAt(i) ||
                "3" == input.charAt(i) ||
                "4" == input.charAt(i) ||
                "5" == input.charAt(i) ||
                "6" == input.charAt(i) ||
                "7" == input.charAt(i) ||
                "8" == input.charAt(i) ||
                "9" == input.charAt(i) ||
                "." == input.charAt(i)){}else{
                notFloat = True;
            }
        }
        return notFloat;
    }

    setDataClass(this){
        if (this.notFloat() == False && this.notInt() == True){
            print("### item {item} 1 - FLOAT", this.notfloat(), this.notint())
            this.inputClass = "FLOAT"
            this.input = parseFloat(this.input)
        }
        else if (this.notFloat() == False && this.notInt() == False){
            print("### item {item} 1 - INTEGER", this.notfloat(), this.notint())
            this.inputClass = "INTEGER"
            this.input = parseInt(this.input)
        }
        else{
            print("### item {item} 1 - STRING", this.notfloat(), this.notint())
            this.inputClass = "STRING"
        }
    }

    get_class(this){
        return this.inputClass;
    }
    get_input(this){
        return this.input;
    }
}

class PointPerCharacterConvertingSystem{
    constructor(input){
        this.input = input;
        this.letters = ["A","Ā","B","C","Č","D","E","Ē","F","G","Ģ","H","I","Ī","J","K","Ķ","L","Ļ","M","N","Ņ","O","P","Q","R","S","Š","T","U","Ū","V","W","X","Y","Z","Ž"];
        this.output = this.stringProcessor();
    }
    stringProcessor(this){
        let points = 0;
        for (let i = 0; i < this.input.length; i++){
            let letter = this.input.charAt(i)
            if ("0" == letter ||
                "1" == letter ||
                "2" == letter ||
                "3" == letter ||
                "4" == letter ||
                "5" == letter ||
                "6" == letter ||
                "7" == letter ||
                "8" == letter ||
                "9" == letter){
                points += parseInt(letter)
                //print(f"+{letter}")
            }
            else if (letter.upper() in this.letters){
                i = 0
                while (letter.upper() != this.letters[i]){
                    i++
                }
                i++
                points = points + i
                //print(f"+{i}")
            }
        }
        return points;
    }
    get_output(this){
        return this.output;
    }
}
class FloatToInt{
    constructor(this, input){
        this.output = parseInt(String(input).replace(".", ""))
    }
    get_output(this){
        return this.output
    }
}
class ComplexIntegerProcessingScript{
    constructor(input){
        this.input = input
        decrementBase = 100000
        while (decrementBase > 0):
            if      (this.input < 0)               {this.makePositive()}
            else if (this.input in range(0, 5))    {this.pascalTriangleRow()}
            else if (this.input in range(6, 25))   {this.ithFibonacciNumber()}
            else if (this.input in range(26, 101)) {this.nextPrimeNumber()}
            else                                   {this.sumDigits()}
            decrementBase--
            //print(this.input, type(this.input))
        this.output = this.input
    }
    
    makePositive(this){
        //print("makePositive")
        this.input = abs(this.input)
    }
    pascalTriangleRow(this){
        //print("pascalTriangleRow")
        if      (this.input == 0) {this.input = 1}
        else if (this.input == 1) {this.input = 11}
        else if (this.input == 2) {this.input = 121}
        else if (this.input == 3) {this.input = 1331}
        else if (this.input == 4) {this.input = 14641}
        else if (this.input == 5) {this.input = 15101051}
    }
    ithFibonacciNumber(this){
        //print("ithFibonacciNumber")
        a = 1
        b = 1
        for _ in range(this.input):
            a, b = b, a + b
        this.input = a
    }
    nextPrimeNumber(this){
        //print("nextPrimeNumber")
        primes = [29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97, 101]
        for (let i = 0; i < this.input.length; i++){
            prime = this.length[i]
            if (this.input < prime){
                this.input = prime
            }
        }
    }
    sumDigits(this){
        //print("sumDigits")
        sum = 0
        while (this.input > 0){
            digit = this.input % 10
            sum = sum + digit
            this.input = this.input / 10
        }
        this.input = parseInt(sum)
    }
    get_output(this){
        return parseInt(this.output)
    }
}
class Processing:
    constructor(this, input):
        this.list = input
        for item in input:
            print(f"# item {item} START")
            this.dataitem = item
            print(f"## item {item} 1 START")
            this.firstStage = InputPrework(this.dataitem)
            print(f"## item {item} 1 COMPLETE")
            print(this.firstStage.get_class(), this.firstStage.get_input())
            if this.firstStage.get_class() == "STRING":
                print(f"## item {item} 2 START")
                this.secondStage = PointPerCharacterConvertingSystem(this.firstStage.get_input())
                print(this.secondStage.get_output())
                print(f"## item {item} 2 COMPLETE")
                print(f"## item {item} 3 START")
                this.thirdStage = ComplexIntegerProcessingScript(this.secondStage.get_output())
            else if this.firstStage.get_class() == "FLOAT":
                print(f"## item {item} 2 START")
                this.secondStage = FloatToInt(this.firstStage.get_input())
                print(f"## item {item} 2 COMPLETE")
                print(f"## item {item} 3 START")
                this.thirdStage = ComplexIntegerProcessingScript(this.secondStage.get_output())
            else:
                print(f"## item {item} 2 PASS")
                print(this.firstStage.get_input())
                print(f"## item {item} 3 START")
                this.thirdStage = ComplexIntegerProcessingScript(this.firstStage.get_input())
            print(f"## item {item} 3 COMPLETE", this.thirdStage.get_output())

with open('./input.json', 'r') as f:
    file = json.load(f)

iteration = Processing(file)

File: py/oop.py
import json

class InputPrework:
    """Šī klase ir atbildīga par ievaddatu pirmapstrādi - saraksta _(list)_ lasīšanu un datu definēšanu."""
    def __init__(self, input) -> None:
        self.input, self.inputClass = str(input), "?"
        #print("### item {item} 1 - INITIALIZED")
        self.setDataClass()

    def notint(self) -> bool:
        """Atgriež, vai tas ir skaitlis, nevis iespējama daļa"""
        notInt:bool = False
        for char in self.input:
            if char == ".":
                notInt = True
                break
        return notInt

    def notfloat(self) -> bool:
        """Atgriež, vai tā ir daļa"""
        notFloat:bool = False
        for char in self.input:
            if char not in ["-", "0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "."]:
                notFloat = True
                break
        return notFloat

    def setDataClass(self) -> None:
        """Nosaka datu tipu"""
        if   self.notfloat() == False and self.notint() == True:
            #print("### item {item} 1 - FLOAT", self.notfloat(), self.notint())
            self.inputClass, self.input = "FLOAT", float(self.input)
        
        elif self.notfloat() == False and self.notint() == False:
            #print("### item {item} 1 - INTEGER", self.notfloat(), self.notint())
            self.inputClass, self.input = "INTEGER", int(self.input)

        else:
            #print("### item {item} 1 - STRING", self.notfloat(), self.notint())
            self.inputClass = "STRING"

    def get_class(self) -> str:
        """Atgriež datu tipu ("STRING", "INTEGER" vai "FLOAT")"""
        return self.inputClass

    def get_input(self) -> str|int|float:
        """Atgriež datu"""
        return self.input

class PointPerCharacterConvertingSystem:
    """Šī klase ir atbildīga par virknes _(string)_ pārveidošanu."""

    def __init__(self, input) -> None:
        self.input:str = input
        self.LETTERS:list[str] = ["A","Ā","B","C","Č","D","E","Ē","F","G","Ģ","H","I","Ī","J","K","Ķ","L","Ļ","M","N","Ņ","O","P","Q","R","S","Š","T","U","Ū","V","W","X","Y","Z","Ž"]
        self.output = self.stringProcessor()
        
    def stringProcessor(self) -> int:
        """Apstrādā teksta datu"""
        points:int = 0
        for letter in self.input:
            if letter.upper() in ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"]:
                points += int(letter)
                #print(f"+{letter}")
            elif letter.upper() in self.LETTERS:
                i = 0
                while letter.upper() != self.LETTERS[i]:
                    i += 1
                i += 1
                points += i
                #print(f"+{i}")
        return points

    def get_output(self) -> int:
        """Atgriež punktus, kuri sanāca pēc virknes apstrādes"""
        return self.output

class FloatToInt:
    """Šī klase pārveido daļu _(float)_ par skaitli _(integer)_."""
    def __init__(self, input) -> None:
        self.output:int = int(str(input).replace(".", ""))

    def get_output(self) -> int:
        """Atgriež iegūto skaitli, bijušo daļu"""
        return self.output
    
class ComplexIntegerProcessingScript:
    """Šī klase apstrādā skaitļus _(integer)_ ar visdažādākām matemātiskām metodēm 100 reizes."""
    def __init__(self, input) -> None:
        self.input:int = input
        decrementBase:int = 100000
        while decrementBase > 0:
            if   self.input < 0:               self.makePositive()
            elif self.input in range(0, 5):    self.pascalTriangleRow()
            elif self.input in range(6, 25):   self.ithFibonacciNumber()
            elif self.input in range(26, 101): self.nextPrimeNumber()
            else:                              self.sumDigits()
            decrementBase -= 1
            #print(self.input, type(self.input))
        self.output:int = self.input
    
    def makePositive(self) -> None:
        """Atgriež veselo skaitli"""
        #print("makePositive")
        self.input = abs(self.input)

    def pascalTriangleRow(self) -> None:
        """Atgriež Paskāla trīsstūra n-1 rindu"""
        #print("pascalTriangleRow")
        if   self.input == 0: self.input = 1
        elif self.input == 1: self.input = 11
        elif self.input == 2: self.input = 121
        elif self.input == 3: self.input = 1331
        elif self.input == 4: self.input = 14641
        elif self.input == 5: self.input = 15101051
    
    def ithFibonacciNumber(self) -> None:
        """Atgriež n-to Fibonači skaitli"""
        #print("ithFibonacciNumber")
        a, b = 1, 1
        for _ in range(self.input):
            a, b = b, a + b
        self.input = a

    def nextPrimeNumber(self) -> None:
        """Atgriež mazāko no pirms _n_ skaitļa"""
        #print("nextPrimeNumber")
        PRIMES:list[int] = [29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97, 101]
        for prime in PRIMES:
            if self.input < prime:
                self.input = prime

    def sumDigits(self) -> None:
        """Atgriež skaitļa ciparu summu"""
        #print("sumDigits")
        sum:int = 0
        while self.input > 0:
            digit = self.input % 10
            sum += digit
            self.input = self.input / 10
        self.input = int(sum)
    
    #def get_output(self) -> int:
    #    """Atgriež beiga skaitli"""
    #    return int(self.output)

class Processing:
    """Šī klase ir atbildīga par visu datu apstrādi."""
    def __init__(self, input):
        self.list:list[str|int|float] = input
        for item in input:
            #print(f"# item {item} START")
            self.dataitem:str|int|float = item
            #print(f"## item {item} 1 START")
            self.firstStage = InputPrework(self.dataitem)
            #print(f"## item {item} 1 COMPLETE")
            #print(self.firstStage.get_class(), self.firstStage.get_input())
            if self.firstStage.get_class() == "STRING":
                #print(f"## item {item} 2 START")
                self.secondStage = PointPerCharacterConvertingSystem(self.firstStage.get_input())
                #print(self.secondStage.get_output())
                #print(f"## item {item} 2 COMPLETE")
                #print(f"## item {item} 3 START")
                self.thirdStage = ComplexIntegerProcessingScript(self.secondStage.get_output())
            elif self.firstStage.get_class() == "FLOAT":
                #print(f"## item {item} 2 START")
                self.secondStage = FloatToInt(self.firstStage.get_input())
                #print(f"## item {item} 2 COMPLETE")
                #print(f"## item {item} 3 START")
                self.thirdStage = ComplexIntegerProcessingScript(self.secondStage.get_output())
            else:
                #print(f"## item {item} 2 PASS")
                #print(self.firstStage.get_input())
                #print(f"## item {item} 3 START")
                self.thirdStage = ComplexIntegerProcessingScript(self.firstStage.get_input())
            #print(f"## item {item} 3 COMPLETE", self.thirdStage.get_output())

with open('./input.json', 'r') as f:
    FILE = json.load(f)

iteration = Processing(FILE)